applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -569,28 +587,16 @@ CrBeamElement3D2N::CalculateDeformationStiffness() const
     Kd(4, 4) = 3.0 * E * Iy * Psi_y / L;
     Kd(5, 5) = 3.0 * E * Iz * Psi_z / L;
 
-    const double l = StructuralMechanicsElementUtilities::CalculateCurrentLength3D2N(*this);
-    const double N = mNodalForces[6];
-
-    const double Qy =
-        -1.00 * (mNodalForces[5] + mNodalForces[11]) / l;
-
-    const double Qz = 1.00 * (mNodalForces[4] + mNodalForces[10]) / l;
 
+    const double N = (E*A/L) * (l-L);

```

Comment:
May be a good idea to put a debug check here to insure that l is minor than L?, even a small check in release to set if l > L l =L; assuming l will be close. this L and l is computed in current frame or reference frame?

Reply:
this is done in   
https://github.com/KratosMultiphysics/Kratos/blob/c375fe7ab64bedd82a62e1bfa4b6169345524476/applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp#L565

calling the utilities

Reply:
Ok